### PR TITLE
Fixes #17166: Remove packaging of ncf api

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/17166

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

File: rudder-webapp/SOURCES/Makefile [hunk 1]
```diff
@@ -183,7 +174,6 @@ install: build initial-promises initial-ncf
 	mkdir -p $(DESTDIR)/usr/share/doc/rudder
 	mkdir -p $(DESTDIR)/usr/share/
 	mkdir -p $(DESTDIR)/usr/bin/
-	mkdir -p $(DESTDIR)/usr/share/ncf-api-virtualenv/share/selinux/
 	mkdir -p $(DESTDIR)/usr/lib/systemd/system/
 	mkdir -p $(DESTDIR)/var/lib/ncf-api-venv/
```
Comment (amousset) [inline rudder-webapp/SOURCES/Makefile:178]: can probably be removed too

File: rudder-webapp/SPECS/rudder-webapp.spec [hunk 2]
```diff
@@ -261,7 +256,6 @@ then
 fi
 
 %if 0%{?suse_version}
-a2enmod rewrite dav dav_fs ssl version wsgi
```
Comment (amousset) [inline rudder-webapp/SPECS/rudder-webapp.spec:264]: we need those (except for wsgi)

Comment (VinceMacBuche) [reply rudder-webapp/SPECS/rudder-webapp.spec:264]: I think i was hacked because i only wanted to remove wsgi !! thanks !!

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
We need some migration steps (delete the user, remose the selinux module, etc.)

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
I deleted user ncf-api-venv and selinux policy is removed by preun of packages  (and not added anymore)

**Comment by VinceMacBuche** [comment]
https://github.com/Normation/rudder-packages/pull/2239/files#diff-905af0b93da870faf3ea7af58142147cR31

**Comment by VinceMacBuche** [comment]
in fact i removed removal of selinux policy, but it will be done in 6.0 post-uninstall 
https://github.com/Normation/rudder-packages/pull/2239/files#diff-6b1d5418abc0c96fb2f3a146d0ebd216L328

**Comment by amousset** [comment]
Ok for me, waiting for @peckpeck

**Comment by amousset** [comment]
+rebase needed

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by amousset** [comment]
OK, squash merging this PR

File: rudder-webapp/SOURCES/rudder-webapp-postinst [hunk 3]
```diff
@@ -27,14 +25,22 @@ fi
...
-  useradd --system --gid ncf-api-venv --shell /bin/false --home-dir /var/lib/ncf-api-venv --groups rudder --comment "ncf API,,," ncf-api-venv >> ${LOG_FILE}
+
+echo -n "INFO: Removing ncf-api-venv user and group ..."
+if getent passwd ncf-api-venv >/dev/null; then
+  userdel ncf-api-venv -r >> ${LOG_FILE}
 fi
-chown -R ncf-api-venv:ncf-api-venv /var/lib/ncf-api-venv
 
+if getent group ncf-api-venv >/dev/null; then
+  groupdel --system ncf-api-venv >> ${LOG_FILE}
```
Comment (ncharles) [inline rudder-webapp/SOURCES/rudder-webapp-postinst:37]: this option fails on SLES12SP4

**Pull request merged**